garak/harnesses/base.py

```diff
@@ -29,7 +29,9 @@ class Harness(Configurable):
 
     active = True
 
-    DEFAULT_PARAMS = {}
+    DEFAULT_PARAMS = {
+        "strict_modality_match": False,
+    }
```

Comment:
This seems reasonable to be in the harness for now, however I see some possible contention as noted in the description questions.

I suspect strict modality matching requirements may lean toward to be a responsibility of the probe to define requirements. I suspect the technique employed in the probe will set the requirement for strict match as generators may often support various modalities.

Reply:
Interested to hear more